resources/sample_vaults/Tasks-Demo/Styling/Styling of Queries.md

```diff
@@ -0,0 +1,40 @@
+# Styling of Queries
+
+To test styling of queries, follow this:
+
+- [ ] **1. Open the Obsidian settings of the Demo vault and under Appearance | CSS Snippets, turn on `tasks-plugin-query-styling`.**
+
+- [ ] **2.** **Test 'group by' classes** - the following query result should have **red headings** named 'Priority 1: High' and 'Priority 4: Low'.
+
+```tasks
+path includes Styling of Queries
+group by priority
+```
```

Comment:
With the code from this PR, this one does not work for  me:

![image](https://user-images.githubusercontent.com/4840096/217620126-e78f18ba-8261-4f4c-a814-d65358caa2e5.png)

Reply:
I did not include in the PR a new compiled JS file, are you using in the test vault a plugin built from this PR and still get black headings?

Reply:
Yes.

Reply:
That's really odd, any clue what might be the difference between your demo vault and mine? To my understanding it's supposed to be a clean slate...
If you inspect the `ul class="contains-task-list"` element like in the screenshot, do you see the `tasks-group-by-priority` class?

Reply:
![2023-02-08_22-02](https://user-images.githubusercontent.com/10722656/217639201-d0ac4c60-06e2-45e3-9b6f-b26247e59d08.png)

Reply:
Yes I do:

![image](https://user-images.githubusercontent.com/4840096/217640713-43fe254b-bce0-4a06-9ca3-12b294ff79fc.png)

I do this to do a local build:

```
rm -f main.js && \
    yarn run build:dev && \
    ./scripts/Test-TasksInLocalObsidian.sh && \
    mdsnippets && yarn run lint:markdown && git add --renormalize . && \
    open "obsidian://open?vault=9a93c6cfdff39622"
```

Other than the change in `appearance.json` to enable the snippet, and the updated `main.js` and `manifest.json` there are no git changes.

Reply:
Ignore the mdsnippets line - it does some processing of docs files and does not affect the behaviour of code.

Reply:
Is it significant that the hearings do not have any classes on?

![image](https://user-images.githubusercontent.com/4840096/217642332-68cc1972-805a-45ef-af6f-011fbfae8c86.png)

Reply:
I am using b7b2287fa034085e83c42547312ceccd8f9eb37f

Reply:
I get the same behaviour with 551bcc99257943042b0d201b3dc5e54d96efeb1a

Reply:
Seems like the code worked (it added the class that it should have) but the CSS snippet didn't do its job.
I gotta call it a day but will try to track your steps and test it the same way tomorrow, hoping to understand the difference between what we're seeing.
Regarding the headings, indeed they don't have any classes on them; It required major changes to the grouping code, and doesn't have any real benefit anyway, as far as they're easily selectable from where the grouping class is.

Reply:
I put some replies to this in the wrong place - they start at https://github.com/obsidian-tasks-group/obsidian-tasks/pull/1519#discussion_r1100657944

Reply:
Picking this up, I used the Theme Design Utilities plugin to find out Tech Stack Versions and it said:

![image](https://user-images.githubusercontent.com/4840096/217665238-26818e7c-89ed-44d4-a792-aabf83bd53dd.png)

I don't know if it's relevant, but Browser Compatibility for `has()` suggests Chrome 105 is required:

https://developer.mozilla.org/en-US/docs/Web/CSS/:has#browser_compatibility

Here is the debug output from my vault (I had initially turned off all plugins except Tasks)

```text
SYSTEM INFO:
	Obsidian version: v1.1.12
	Installer version: v0.15.9
	Operating system: Darwin Kernel Version 19.6.0: Mon Aug 31 22:12:52 PDT 2020; root:xnu-6153.141.2~1/RELEASE_X86_64 19.6.0
	Login status: logged in
	Catalyst license: vip
	Insider build toggle: on
	Live preview: on
	Legacy editor: off
	Base theme: light
	Community theme: none
	Snippets enabled: 1
	Restricted mode: off
	Plugins installed: 5
	Plugins enabled: 2
		1: Tasks v1.24.0
		2: Theme Design Utilities v0.21.1

RECOMMENDATIONS:
	Custom theme and snippets: for cosmetic issues, please first try updating your theme and disabling your snippets. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community theme and snippets.
	Community plugins: for bugs, please first try updating all your plugins to latest. If still not fixed, please try to make the issue happen in the Sandbox Vault or disable community plugins.
```

I wonder whether my out-of-date installer might be the difference in behaviour?

Reply:
Yay for smoke tests! 😆

Reply:
Now what I don't know is what the minimum installer version is that we should target...

Reply:
I have documented in the `styling.md` and the smoke-test file that at least Obsidian 1.1.9 installer is required.

It's very slightly possible that a slightly earlier installer may also work, but I just don't have the time to spend tracking that down.